pkgs/development/libraries/kerberos/krb5.nix

```diff
@@ -83,34 +83,32 @@ stdenv.mkDerivation rec {
 
   libFolders = [ "util" "include" "lib" "build-tools" ];
 
-  buildPhase = lib.optionalString libOnly ''
-    runHook preBuild
-
-    MAKE="make -j $NIX_BUILD_CORES"
-    for folder in $libFolders; do
-      $MAKE -C $folder
+  # To avoid cyclic outputs, we can't let lib depend on out in any way. Unfortunately, the configure
+  # options don't give us enough granularity to specify that, so we have to override the generated
+  # Makefiles manually.
+  postConfigure = ''
+    find $libFolders -type f -name Makefile -print0 | while IFS= read -rd "" f; do
+      substituteInPlace "$f" --replace-fail "$out" "$lib"
     done
-
-    runHook postBuild
   '';
 
-  installPhase = lib.optionalString libOnly ''
-    runHook preInstall
-
-    mkdir -p "$out"/{bin,sbin,lib/pkgconfig,share/{et,man/man1}} \
-      "$dev"/include/{gssapi,gssrpc,kadm5,krb5}
-    for folder in $libFolders; do
-      $MAKE -C $folder install
-    done
-
-    runHook postInstall
+  preInstall = ''
+    mkdir -p "$lib"/{bin,sbin,lib/pkgconfig,share/{et,man/man1}}
   '';
```

Comment:
So basically the original derivation was forcibly overriding a bunch of standard steps to avoid compiling certain parts that would have resulted in a circular dependency, right?

Reply:
Like I said in another comment, I think the original derivation was trying to avoid `libedit` and `bison` and it found a way to do that by just building the shared objects. But we can avoid them differently.

It's also fair to say that the original derivation was optimized for the common case: `libkrb5` is used far more than `krb5`. But, that causes issues as I describe in the PR description.